src/test/scala/io/iohk/ethereum/network/PeerEventBusActorSpec.scala

```diff
@@ -46,6 +56,33 @@ class PeerEventBusActorSpec extends AnyFlatSpec with Matchers {
     peerEventBusActor ! PeerEventBusActor.Publish(msgFromPeer2)
     probe1.expectNoMessage()
     probe2.expectMsg(msgFromPeer2)
+
+  }
+
+  it should "relay messages via streams" in new TestSetup {
+    val classifier1 = MessageClassifier(Set(Ping.code), PeerSelector.WithId(PeerId("1")))
+    val classifier2 = MessageClassifier(Set(Ping.code), PeerSelector.AllPeers)
+
+    val completeOnTermination =
+      Flow[MessageFromPeer].recoverWithRetries(1, { case _: WatchedActorTerminatedException => Source.empty })
+
+    val stream1 =
+      PeerEventBusActor.messageSource(peerEventBusActor, classifier1).via(completeOnTermination).runWith(Sink.seq)
+    val stream2 =
+      PeerEventBusActor.messageSource(peerEventBusActor, classifier2).via(completeOnTermination).runWith(Sink.seq)
+
+    Thread.sleep(100) // stream is not subscribed right away
```

Comment:
I managed to remove the sleep with a probe that forwards messages, and adding an `expectMsgType[Subscribe]` before publishing.

I don't know if it is a really good idea, but it seems to fit the bill to allow waiting for subscribe without modifying production code only for that purpose.
  
```scala
    val classifier1 = MessageClassifier(Set(Ping.code), PeerSelector.WithId(PeerId("1")))
    val classifier2 = MessageClassifier(Set(Ping.code), PeerSelector.AllPeers)
    val peerEventBusProbe = TestProbe()(system)
    peerEventBusProbe.setAutoPilot { (sender: ActorRef, msg: Any) =>
      peerEventBusActor.tell(msg, sender)
      TestActor.KeepRunning
    }

    val completeOnTermination =
      Flow[MessageFromPeer].recoverWithRetries(1, { case _: WatchedActorTerminatedException => Source.empty })

    val stream1 =
      PeerEventBusActor.messageSource(peerEventBusProbe.ref, classifier1).via(completeOnTermination).runWith(Sink.seq)

    peerEventBusProbe.expectMsgType[Subscribe]
    val stream2 =
      PeerEventBusActor.messageSource(peerEventBusProbe.ref, classifier2).via(completeOnTermination).runWith(Sink.seq)
    peerEventBusProbe.expectMsgType[Subscribe]

    val msgFromPeer = MessageFromPeer(Ping(), PeerId("1"))
    peerEventBusActor ! PeerEventBusActor.Publish(msgFromPeer)

    val msgFromPeer2 = MessageFromPeer(Ping(), PeerId("99"))
    peerEventBusActor ! PeerEventBusActor.Publish(msgFromPeer2)

    peerEventBusProbe.ref ! PoisonPill

    whenReady(stream1)(_ shouldEqual Seq(msgFromPeer))
    whenReady(stream2)(_ shouldEqual Seq(msgFromPeer, msgFromPeer2))
 ```

Reply:
Ohh that is clever, I like it!